lib/logflare/source/bigquery/buffer_producer.ex

```diff
@@ -33,13 +33,8 @@ defmodule Logflare.Source.BigQuery.BufferProducer do
 
   @spec ack(atom(), [%Broadway.Message{}], [%Broadway.Message{}]) :: :ok
   def ack(source_id, successful, unsuccessful) when is_atom(source_id) do
-    Enum.each(successful, fn %{data: %LE{}} = message ->
-      BufferCounter.ack(source_id, message.data.id)
-    end)
-
-    Enum.each(unsuccessful, fn %{data: %LE{}} = message ->
-      BufferCounter.ack(source_id, message.data.id)
-    end)
+    BufferCounter.ack(source_id, successful)
+    BufferCounter.ack(source_id, unsuccessful)
```

Comment:
Doing `BufferCounter.ack(source_id, successful ++ unsuccessful)` would work too. or `BufferCounter.ack(source_id, {successful, unsuccessful})` if splitting the counts has value in future

Reply:
Good call, changed.